dspace-api/src/main/java/org/dspace/content/service/FindableObjectService.java

```diff
@@ -10,23 +10,20 @@
 import java.io.Serializable;
 import java.sql.SQLException;
 
-import org.dspace.browse.BrowsableObject;
 import org.dspace.core.Context;
 
 /**
- * Service interface class for any BrowsableDSpaceObject.
- * All BrowsableObject service classes should implement this class since it offers some basic methods which all
- * BrowsableObjects are required to have.
+ * Base Service interface class for any Persistent Entity findable by a primary key.
  *
- * @param <T> class type
+ * @param <T> class type of the persistent entity
+ * @param <PK> class type of the primary key
  * @author Andrea Bollini (andrea.bollini at 4science.it)
  */
-public interface BrowsableObjectService<T extends BrowsableObject<PK>, PK extends Serializable> {
+public interface FindableObjectService<T, PK extends Serializable> {
```

Comment:
Should the T here not extend the IndexableDSpaceObject ?

Reply:
No as it can have a more broad adoption, i.e. be implemented by any persistence service

Reply:
Due to the agreement in https://github.com/DSpace/DSpace/pull/2391#discussion_r273412647 the interface is now just limited to IndexableObject so I have applied your suggestion as well